### DISCUSSION TITLE
May I get some help on this SessionStorage code?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#57957] [Help]
### Summary

What am I looking to do?

1. Create a sessionStorage item and save it in the browswer.
2. At the end of the code line I want to check every second for the item in the sessionStorage.
3.  If the item in the sessionStorage  is not found, and the .length of another item has a certain value, then I want to trigger a function.

```
// These lines below set the sessionStorage item, which I think is working, and sets the values of the item I want to email.

const conversationId =
    selectedConversationId === null ? uuid() : selectedConversationId;

  const sessionIdStorage = (e) => {
    e.preventDefault();
    localStorage.setItem("sessionId", conversationId);
  };

  const sessionId = () => {
    localStorage.getItem(sessionIdStorage());
  };

  let chatToServer;

  const history = chatHistory.find((c) => c.id === conversationId);
  if (history) {
    chatToServer = history.messages;
  } else {
    chatToServer = [];
  }

  let chatToEmailServer = {
    chatToServer,
  };

//  Then at the very bottom of the code I want to check for the sessionStorage item and if there is a something in the array value of the item I want.

 setInterval(() => {
    if (sessionId === null) {
      if (chatToServer.length > 0) {
        emailChatHistory();
      }
    }
  }, 1000);
```

^^^ So far this code isn't doing anything, but the idea is to trigger a function once the tab in the browser is closed.

**Comment by BohdanYavorskyi** [comment]
is your setInterval inside `useEffect`?

**Reply by Elindo586** [reply]
Oh it is not...  let me check what happens if I put it in a `useEffect`

**Reply by BohdanYavorskyi** [reply]
Besides that fact it won't work as well I guess. You have to catch updates to your dependencies which are `sessionId`,`chatToServer.length` and `emitChatHistory`.

Do it like:

```
const intervalId = setInterval(yourHandler);

return () => clearInterval(id);

```
So effect will recreate your interval func when dependencies are updated

**Reply by Elindo586** [reply]
Apparently when I want to add dependencies it gives me an error...   also....    assuming this would work I still need to find out how to kill the interval properly, else it would keep executing the function over and over and I don't want that.... maybe add a true false value somewhere to kill the interval..

**Reply by BohdanYavorskyi** [reply]
@Elindo586 It will kill it each time the dependency changes... That's what I mentioned above

**Reply by Elindo586** [reply]
Yes....  say that I can add the `chatToServer`  in the dependencies array (right now I am getting errors anyway) .... the `chatToServer.length`  is going to grow and change....   I don't want to execute the function when that item changes......   I only want to execute the function  when the sessionId is no longer in the sessionStorage in the browser... aka when the browser tab is closed...

**Reply by BohdanYavorskyi** [reply]
Using conditions you'll achieve that. The interval logic is same
